definitions/scenarios/upgrade_to_satellite_6_13.rb

```diff
@@ -0,0 +1,95 @@
+module Scenarios::Satellite_6_13
+  class Abstract < ForemanMaintain::Scenario
+    def self.upgrade_metadata(&block)
+      metadata do
+        tags :upgrade_scenario
+        confine do
+          feature(:satellite) &&
+            (feature(:satellite).current_minor_version == '6.12' || \
+            ForemanMaintain.upgrade_in_progress == '6.13')
+        end
+        instance_eval(&block)
+      end
+    end
+
+    def target_version
+      '6.13'
+    end
+  end
+
+  class PreUpgradeCheck < Abstract
+    upgrade_metadata do
+      description 'Checks before upgrading to Satellite 6.13'
+      tags :pre_upgrade_checks
+      run_strategy :fail_slow
+    end
+
+    def compose
+      add_steps(find_checks(:default))
+      add_steps(find_checks(:pre_upgrade))
+      add_step(Checks::Foreman::CheckpointSegments)
+      add_step(Checks::Repositories::Validate.new(:version => '6.13'))
+    end
+  end
+
+  class PreMigrations < Abstract
+    upgrade_metadata do
+      description 'Procedures before migrating to Satellite 6.13'
+      tags :pre_migrations
+    end
+
+    def compose
+      add_steps(find_procedures(:pre_migrations))
+      add_step(Procedures::Service::Stop.new)
+    end
+  end
+
+  class Migrations < Abstract
+    upgrade_metadata do
+      description 'Migration scripts to Satellite 6.13'
+      tags :migrations
+      run_strategy :fail_fast
+    end
+
+    def set_context_mapping
+      context.map(:assumeyes, Procedures::Installer::Upgrade => :assumeyes)
+    end
+
+    def compose
+      add_step(Procedures::Repositories::Setup.new(:version => '6.13'))
+      modules_to_enable = ["satellite:#{el_short_name}"]
+      add_step(Procedures::Packages::EnableModules.new(:module_names => modules_to_enable))
+      add_step(Procedures::Packages::UnlockVersions.new)
+      add_step(Procedures::Packages::Update.new(:assumeyes => true))
+      add_step_with_context(Procedures::Installer::Upgrade)
+      add_step(Procedures::Installer::UpgradeRakeTask)
+    end
+  end
+
+  class PostMigrations < Abstract
+    upgrade_metadata do
+      description 'Procedures after migrating to Satellite 6.13'
+      tags :post_migrations
+    end
+
+    def compose
+      add_step(Procedures::RefreshFeatures)
+      add_step(Procedures::Service::Start.new)
+      add_step(Procedures::Pulpcore::TrimRpmChangelogs.new)
```

Comment:
We originally said we would not do it on the "to 6.12" upgrade, but as we didn't ship it in 6.12 yet… I left it in.